oauthproxy.go

```diff
@@ -90,6 +93,7 @@ type OAuthProxy struct {
 
 	redirectURL             *url.URL // the url to receive requests at
 	whitelistDomains        []string
+	extraHeaders            [][]string
```

Comment:
You could use an `http.Header` here instead, then you wouldn't need to index the strings when setting them on the request
```
extraHeaders := make(http.Header)
for _, extraHeader := range opts.ExtraHeaders {
  match := ...
  extraHeaders.Add(match[1], match[2])
}
```

```
for key, values := range p.extraHeaders {
  req.Header.Set/Add(key, strings.Join(", ", values...)
}
```